File: hbTapi Components/BCB Samples/Professional Edition/AddressTranslation/uMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "uMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "hbTapi"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::FormCreate(TObject *Sender)
{
  ComboBox_Device->Items->Assign(hbTapiLine1->DeviceList);
  if (ComboBox_Device->Items->Count > 0)
    ComboBox_Device->ItemIndex = 0;
  ComboBox_DeviceChange(ComboBox_Device);
  ComboBox_Options->ItemIndex = 0;

}
//---------------------------------------------------------------------------

void __fastcall TForm1::ComboBox_DeviceChange(TObject *Sender)
{
  hbTapiLine1->DeviceID = ComboBox_Device->ItemIndex;
  ComboBox_PhoneNo->Enabled = hbTapiLine1->Available;
  ComboBox_Options->Enabled = hbTapiLine1->Available;

}

//---------------------------------------------------------------------------
void __fastcall TForm1::Button_TranslateDialogClick(TObject *Sender)
{
  hbTapiLine1->TranslateDialog(ComboBox_PhoneNo->Text, Handle);
}

//---------------------------------------------------------------------------
void __fastcall TForm1::Button_TranslateAddressClick(TObject *Sender)
{
  unsigned o;
  switch (ComboBox_PhoneNo->ItemIndex) {
    case 1 :
      o = LINETRANSLATEOPTION_CARDOVERRIDE;
      break;
    case 2 :
      o = LINETRANSLATEOPTION_CANCELCALLWAITING;
      break;
    case 3 :
      o = LINETRANSLATEOPTION_FORCELOCAL;
      break;
    case 4 :
      o = LINETRANSLATEOPTION_FORCELD;
      break;
    default :
     o = 0;
  }
  MessageDlg(hbTapiLine1->TranslateAddress(ComboBox_PhoneNo->Text, o), mtInformation, TMsgDlgButtons() << mbOK, 0);

}
//---------------------------------------------------------------------------

File: hbTapi Components/Delphi Samples/Professional Edition/AddressTranslation/uMain.pas
{-------------------------------------------------------------------------
   Copyright by Haeger + Busch, Germany                / >>>>>>>>> /-----
   Ingenieurbuero fuer Kommunikationslösungen         / <<<<<<<<< /
 ----------------------------------------------------/ >>>>>>>>> /
   Homepage  : http://www.hbTapi.com
   EMail     : info@hbTapi.com
   Package   : hbTapi Components
 -------------------------------------------------------------------------}
unit uMain;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  hbTAPI, hbTapiApi, StdCtrls;

type
  TForm1 = class(TForm)
    ComboBoxDevice: TComboBox;
    ComboBoxPhoneNo: TComboBox;
    ButtonTranslateDlg: TButton;
    ButtonTranslateAddress: TButton;
    hbTapiLine1: ThbTapiLine;
    ComboBoxOptions: TComboBox;
    Label1: TLabel;
    Label2: TLabel;
    Label3: TLabel;
    procedure FormCreate(Sender: TObject);
    procedure ComboBoxDeviceChange(Sender: TObject);
    procedure ButtonTranslateDlgClick(Sender: TObject);
    procedure ButtonTranslateAddressClick(Sender: TObject);
  private
    { Private-Deklarationen }
  public
    { Public-Deklarationen }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.FormCreate(Sender: TObject);
begin
  ComboBoxDevice.Items.Assign(hbTapiLine1.DeviceList);
  if ComboBoxDevice.Items.Count > 0 then
    ComboBoxDevice.ItemIndex := 0;
  ComboBoxDeviceChange(ComboBoxDevice);
  ComboBoxOptions.ItemIndex := 0;
end;

procedure TForm1.ComboBoxDeviceChange(Sender: TObject);
begin
  hbTapiLine1.DeviceID := ComboBoxDevice.ItemIndex;
  ComboBoxPhoneNo.Enabled := hbTapiLine1.Available;
  ComboBoxOptions.Enabled := hbTapiLine1.Available;
end;

procedure TForm1.ButtonTranslateDlgClick(Sender: TObject);
begin
  hbTapiLine1.TranslateDialog(ComboBoxPhoneNo.Text, Handle);
end;

procedure TForm1.ButtonTranslateAddressClick(Sender: TObject);
var o: DWORD;
begin
 case ComboBoxOptions.ItemIndex of
   1 : o := LINETRANSLATEOPTION_CARDOVERRIDE;
   2 : o := LINETRANSLATEOPTION_CANCELCALLWAITING;
   3 : o := LINETRANSLATEOPTION_FORCELOCAL;
   4 : o := LINETRANSLATEOPTION_FORCELD;
 else
   o := 0;
 end;
 MessageDlg(hbTapiLine1.TranslateAddress(ComboBoxPhoneNo.Text, o), mtInformation, [mbOK], 0);
end;

end.
